nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -58,14 +62,11 @@ let
         version extraConfig extraPerEntryConfig extraEntries forceInstall useOSProber
         extraEntriesBeforeNixOS extraPrepareConfig extraInitrd configurationLimit copyKernels
         default fsIdentifier efiSupport efiInstallAsRemovable gfxmodeEfi gfxmodeBios;
-      path = (makeBinPath ([
-        pkgs.coreutils pkgs.gnused pkgs.gnugrep pkgs.findutils pkgs.diffutils pkgs.btrfs-progs
-        pkgs.utillinux ]
-        ++ (optional (cfg.efiSupport && (cfg.version == 2)) pkgs.efibootmgr)
-        ++ (optionals cfg.useOSProber [pkgs.busybox pkgs.os-prober])
-      )) + ":" + (makeSearchPathOutput "bin" "sbin" [
-        pkgs.mdadm pkgs.utillinux
-      ]);
+      path = with pkgs; (makeBinPath (
+        [ coreutils gnused gnugrep findutils diffutils btrfs-progs utillinux ]
+        ++ (optional (cfg.efiSupport && (cfg.version == 2)) efibootmgr)
+        ++ (optionals cfg.useOSProber [ busybox os-prober ])
+      )) + ":" + (makeSearchPathOutput "bin" "sbin" [ mdadm utillinux ]);
```

Comment:
One nitpicky thing: There's a whole bunch of unnecessary parens here, can be this instead:
```nix
{
  path = with pkgs; makeBinPath (
    [ coreutils gnused gnugrep findutils diffutils btrfs-progs utillinux ]
    ++ optional (cfg.efiSupport && cfg.version == 2) efibootmgr
    ++ optionals cfg.useOSProber [ busybox os-prober ]
  ) + ":" + makeSearchPathOutput "bin" "sbin" [ mdadm utillinux ];
}
```

And considering that there's symlinks from `sbin -> bin` in `mdadm` and `utillinux`, this can be refactored to
```nix
{
  path = with pkgs; makeBinPath (
    [ coreutils gnused gnugrep findutils diffutils btrfs-progs utillinux mdadm utillinux ]
    ++ optional (cfg.efiSupport && cfg.version == 2) efibootmgr
    ++ optionals cfg.useOSProber [ busybox os-prober ]);
}
```

Reply:
The reason to change this line in the first place was to make the expression more readable, so you should be nitpicky about the result.

I am not sure about the implications of no longer including the ```sbin``` folder of ```mdadm``` and ```utillinux``` in the path, so I will just take your word for it, as I lack the time for testing at the moment.

Please see the latest version of the patch for the result.

Reply:
I checked, sbin of `mdadm` and `utillinux` was just a symlink to the `bin` directory, so `makeSearchPathOutput "bin" "sbin" [ mdadm utillinux ] = makeSearchPathOutput "bin" "bin" [ mdadm utillinux ] = makeBinPath [ mdadm utillinux ]`